django/middleware/csrf.py

```diff
@@ -196,6 +197,37 @@ def _set_token(self, request, response):
             # Set the Vary header since content varies with the CSRF cookie.
             patch_vary_headers(response, ('Cookie',))
 
+    def _origin_verified(self, request):
+        good_origin = '%s://%s' % (
+            'https' if request.is_secure() else 'http',
+            request.get_host(),
+        )
+        allowed_origins = [good_origin]
+        # Tuples of (allowed scheme, allowed netloc), where all
+        # subdomains of netloc are allowed.
+        allowed_origin_subdomains = []
+        for trusted_origin in settings.CSRF_TRUSTED_ORIGINS:
+            if '*' in trusted_origin:
+                parsed = urlparse(trusted_origin)
+                allowed_origin_subdomains.append(
+                    (parsed.scheme, parsed.netloc.lstrip('*'))
+                )
+            else:
+                allowed_origins.append(trusted_origin)
```

Comment:
This just occurred to me, but would it make sense to structure things so the `CSRF_TRUSTED_ORIGINS` processing can be done just once instead of with each request? (There is similar parsing / processing [below](https://github.com/django/django/pull/13829/files#diff-eaa105f5b436e20dd838c27c7a753ef4cf888edcc8f868c084600f6cb7343166R314-R317) in the referer checking.)

Reply:
Yeah the settings dependent part could be moved into the `__init__` of the middleware I guess.

Reply:
I used `@cached_property` so they are only computed if needed.